### PR TITLE
Update outcome

### PR DESCRIPTION
**Pull request by beckerhe**
This is updating outcome to version 2.2.0 and this involves a couple of things:

- The `OUTCOME_TRY` macro requires now a type declaration which allows now to move the value out of the temporary `outcome::result` value. In short `OUTCOME_TRY(result, expr())` becomes `OUTCOME_TRY(auto&& result, expr())`
- Each version of `outcome` puts their symbols into a versioned namespace to avoid ABI issues. To make outcome available in the `outcome` namespace a namespace alias declaration is needed. We used to have that alias as part of the outcome package. But since we are now moving to the official outcome package we need to make put that declaration into Orbit's code. I created a separate module `src/Outcome` which contains a single header file. This header needs to be included instead of including `outcome.hpp` directly.

Unfortunately this all needs to be in a single commit.

### CODE REVIEW

File: src/Outcome/include/Outcome/Outcome.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
```
Comment (dimitry-) [inline src/Outcome/include/Outcome/Outcome.h:1]: can we include this in OrbitBase/Result.h?

Comment (beckerhe) [reply src/Outcome/include/Outcome/Outcome.h:1]: If we want that, yes. I wasn't sure if we want to mix that.

Comment (dimitry-) [reply src/Outcome/include/Outcome/Outcome.h:1]: it is effectively already very much tied to outcome. we use `outcome::success()` everywhere.

Comment (beckerhe) [reply src/Outcome/include/Outcome/Outcome.h:1]: Okay, so I will move it there. Thanks, Dimitry!

Comment (beckerhe) [reply src/Outcome/include/Outcome/Outcome.h]: Done.

**Review by dimitry- — APPROVED** [review]

**Comment by antonrohr** [comment]
Was about to LGTM.. anyways. Thanks for the work!

**Comment by beckerhe** [comment]
> Was about to LGTM.. anyways. Thanks for the work!

Ah. I'm sorry.

**Pull request merged**